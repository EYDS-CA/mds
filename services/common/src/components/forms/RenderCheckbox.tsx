import React, { FC, useContext } from "react";
import { Checkbox, Form } from "antd";
import { BaseInputProps } from "./BaseInput";
import { FormContext } from "./FormWrapper";
/**
 * @constant RenderCheckbox - Ant Design `Checkbox` component for redux-form.
 */

interface CheckboxProps extends BaseInputProps {
  label: string;
}

const RenderCheckbox: FC<CheckboxProps> = ({
  input,
  meta,
  id,
  disabled = false,
  label,
  ...props
}) => {
  const onChange = (e) => {
    input.onChange(e.target.checked);
  };
  const { isEditMode } = useContext(FormContext);

  if (!isEditMode) {
    return (
      <Form.Item
        name={input.name}
        className="view-item"
      >
        <Checkbox id={id} checked={input.value} disabled={true}>
          <div className="view-item-label ungroup-checkbox">{label}</div>
        </Checkbox>
      </Form.Item>
    );
  }
  return (
    <Form.Item
      name={input.name}
      validateStatus={meta.touched ? meta.error && "error" : ""}
      required={props.required}
      className={props.required && "ant-form-item-checkbox-required"}
      help={
        meta.touched &&
        ((meta.error && <span>{meta.error}</span>) || (meta.warning && <span>{meta.warning}</span>))
      }
    >
      <Checkbox id={id} checked={input.value} disabled={disabled} onChange={onChange}>
        {label}
      </Checkbox>
    </Form.Item>
  );
};

export default RenderCheckbox;
