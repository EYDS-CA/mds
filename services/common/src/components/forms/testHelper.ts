export const inputMeta = {
    touched: false,
    error: false,
    warning: false,
    autofilled: false,
    asyncValidating: false,
    dirty: false,
    dispatch: jest.fn(),
    form: "FORM",
    initial: undefined,
    invalid: false,
    pristine: true,
    submitFailed: false,
    submitting: false,
    valid: true,
    visited: false,
};

export const errorInputMeta = {
    touched: true,
    error: "Error message",
    warning: false,
    autofilled: false,
    asyncValidating: false,
    dirty: true,
    dispatch: jest.fn(),
    form: "FORM",
    initial: undefined,
    invalid: true,
    pristine: true,
    submitFailed: false,
    submitting: false,
    valid: true,
    visited: true,
}

// needs a value and a name
export const inputProps = {
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onDragStart: jest.fn(),
    onDrop: jest.fn(),
    onFocus: jest.fn()
}