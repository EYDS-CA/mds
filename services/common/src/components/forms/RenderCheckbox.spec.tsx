import React from "react";
import { render } from "@testing-library/react";
import RenderCheckbox from "./RenderCheckbox";
import { inputMeta, inputProps } from "./testHelper";

describe("RenderCheckbox", () => {
    it("renders properly", () => {
        const { container } = render(<RenderCheckbox
            id="check"
            label="Mock Label"
            input={{ value: "", name: "check", ...inputProps }}
            meta={inputMeta}
        />);
        expect(container).toMatchSnapshot();
    })
})