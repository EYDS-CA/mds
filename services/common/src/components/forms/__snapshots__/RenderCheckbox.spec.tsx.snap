// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderCheckbox renders properly 1`] = `
<div>
  <div
    class="ant-form-item ant-form-item-with-help"
  >
    <div
      class="ant-row ant-form-item-row"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <label
              class="ant-checkbox-wrapper ant-checkbox-wrapper-in-form-item"
            >
              <span
                class="ant-checkbox"
              >
                <input
                  class="ant-checkbox-input"
                  id="check"
                  type="checkbox"
                  value=""
                />
                <span
                  class="ant-checkbox-inner"
                />
              </span>
              <span>
                Mock Label
              </span>
            </label>
          </div>
        </div>
        <div
          style="display: flex; flex-wrap: nowrap;"
        >
          <div
            class="ant-form-item-explain ant-form-item-explain-connected"
            id="check_help"
            role="alert"
          >
            <div
              class=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
